core/src/test/scala/org/apache/spark/scheduler/EventLoggingListenerSuite.scala

```diff
@@ -97,6 +97,17 @@ class EventLoggingListenerSuite extends SparkFunSuite with LocalSparkContext wit
     assert(redactedProps(key) == "*********(redacted)")
   }
 
+  test("Spark-33504 password redaction in properties") {
+    val key = "spark.executorEnv.HADOOP_CREDSTORE_PASSWORD"
+    val secretPassword = "secret_password"
+    val conf = getLoggingConf(testDirPath, None).set(key, secretPassword)
+    val properties = new Properties()
+    properties.setProperty(key, secretPassword)
+    val eventLogger = new EventLoggingListener("test", None, testDirPath.toUri(), conf)
+    val redactedProperties = eventLogger.redactProperties(properties)
```

Comment:
Test whether `SparkListenerJobStart` is updated as required ? `redactProperties` can be made private